phoenix-ios/phoenix-ios/security/InvalidPin.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+struct InvalidPin: Equatable, Codable {
+	let count: Int
+	let timestamp: Date
+	
+	static func none() -> InvalidPin {
+		return InvalidPin(count: 0, timestamp: Date.distantPast)
+	}
+	
+	func increment() -> InvalidPin {
+		let newCount = (count < Int.max) ? count + 1 : Int.max
+		let newTimestamp = Date.now
+		
+		return InvalidPin(count: newCount, timestamp: newTimestamp)
+	}
+	
+	var waitTime: TimeInterval { // TimeInterval => seconds (Double)
+		switch count {
+			case 0  : return  0.seconds()
+			case 1  : return  0.seconds()
```

Comment:
I think we could allow 3 quick failed attempts? or make the third attempt wait 10s.

Also, how long does it take for the interval to reset? E.g, I made 5 failed attempts. If I wait 1 day and try again, does the failed attempts count reset to 0?

Reply:
> If I wait 1 day and try again, does the failed attempts count reset to 0?

Currently, no. But that's a really good idea. I'll add it.

Reply:
Done in https://github.com/ACINQ/phoenix/pull/560/commits/0218ed64c873c873dfb066b9a0ab30cae35ba790